### PR TITLE
feat(app-config): pwa validation

### PR DESCRIPTION
**Pull request by Francois-Esquire**
Uniform validation for `appConfig`, starting with PWA config. The old PWA config validation has been removed in favor of the `joi` schema.

## Description

Using validation library `joi`, a PWA schema was created for that portion of `appConfig` and wrapped with a validation function `validatePWAConfig`.

Files added:
- validation schemas under `src/server/utils/validation` + tests.

Files removed:
- `src/server/middleware/pwa/validation.js` + test

Files modified (w/ tests if applicable):
- `src/server/utils/onModuleLoad.js`
- `src/server/middleware/pwa/config.js`

## Motivation and Context

Utilize a single point of validation for user specific settings that plug into and configure the app.

## How Has This Been Tested?

Testing suite

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [x] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

Richer error messages that identify problem areas in `appConfig`

### CODE REVIEW

File: src/server/utils/validation/index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      Joi.string().hostname(),
+      Joi.string().pattern(/(?:https?)?[a-z.-]/)
+    )
+  ),
+  configureRequestLog: Joi.function(),
+  extendSafeRequestRestrictedAttributes: safeRequestSchema,
+  pwa: pwaSchema,
+  createSsrFetch: Joi.function(),
+  eventLoopDelayThreshold: Joi.number().allow(Infinity),
+  appCompatibility: semverExtension.semver().valid(),
```
Comment (JamesSingleton) [inline src/server/utils/validation/index.js:43]: Doesn't this come in as 

```json
"app": {
  "compatibility : "x.x.x"
}
```

I forget where the conversion to `appCompatibility` comes into play.

Comment (Francois-Esquire) [reply src/server/utils/validation/index.js]: Yes! `@americanexpress/one-app-bundler` actually sets this on the modules meta data. When we come to validate it, we can access it from the module and make sure it's compatible with the running One App instance.

File: __tests__/server/utils/validation/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        escapeHatch: 0,
+        recoveryMode: [],
+        scope: '\\',
+        webManifest: { short_name: 'One App Test' },
+      })
+    ).toThrow(
+      new Error([
+        '"recoveryMode" must be a boolean',
+        '"escapeHatch" must be a boolean',
+        '"scope" with value "\\" fails to match the required pattern: /^\\//',
```
Comment (mtomcal) [inline __tests__/server/utils/validation/index.spec.js:52]: Can we get a more humanized message for `scope`? https://joi.dev/api/?v=17.2.1#anyruleoptions

File: src/server/utils/validation/webManifest.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+// eslint-disable-next-line import/prefer-default-export
+export const webManifestSchema = Joi.object().keys({
+  name: Joi.string().required(),
+  short_name: Joi.string(),
+  description: Joi.string(),
+  lang: Joi.string(),
+  scope: Joi.string().pattern(/^\//),
+  start_url: Joi.alternatives(
+    Joi.string().uri(),
+    Joi.string().pattern(/^\//)
```
Comment (mtomcal) [inline src/server/utils/validation/webManifest.js:28]: Some more humanized messages for `pattern` would be good.

**Comment by Francois-Esquire** [comment]
@mtomcal I've updated the error messages for `scope` and `start_url` ( a040edc ),  it's a friendlier message.  I've also added more samples to the test to view the output; let me know if we can write/communicate a specific field better.

**Review by infoxicator — APPROVED** [review]

**Pull request merged**